src/lib.rs

```diff
@@ -1393,7 +1405,10 @@ impl Build {
                                 // sure users always see all the compilation failures.
                                 has_made_progress.set(true);
 
-                                let _ = writeln!(stdout, "cargo:warning={}", err);
+                                if self.cargo_warnings {
+                                    let _ = write!(stdout, "cargo:warning=");
+                                }
+                                let _ = writeln!(stdout, "{}", err);
```

Comment:
You should move this inside the `if`, IMO you could actually write a function taking `fmt::Arguments<'_>` and then have a macro to wrap it

Reply:
Just to be clear, you'd prefer that `cargo_warnings(false)` fully disables the output of warnings, not just the 'cargo:warning' prefix?
I don't mind either way.

Reply:
> Just to be clear, you'd prefer that `cargo_warnings(false)` fully disables the output of warnings, not just the 'cargo:warning' prefix? I don't mind either way.

Yes, the output is there as warning, so if the user disables the warning we should also discard the output.